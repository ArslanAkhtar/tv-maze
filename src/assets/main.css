.v-main{
  background-color: black;
  min-height: 90vh;
}


.movie-summary {
  display: -webkit-box;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
  overflow: hidden;
}

.genre {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: flex-start;
}

.loading {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
}

.show-detail {
  display: flex;
  align-items: baseline;
  gap: 5px;
  margin: 10px 0;
}

.loadMore {
  width: 100%;
  display: flex;
  justify-content: center;
}

.no-data-found {
  width: 100%;
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
  margin-bottom: 40px;
}

.page-not-found {
  display: flex;
  flex-direction: column;
  align-items: center;
  flex-wrap: wrap;
  justify-content: center;
  height: 100%;
  
}