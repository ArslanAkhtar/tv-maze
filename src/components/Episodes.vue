<script setup lang="ts">
	import useFetchShows from "@/composables/FetchData";
	import Loader from "@/components/Loader.vue";
	import { imgURL, openExternalURL } from "@/helpers/utility";
	import Error from "@/components/Error.vue";

	const props = defineProps({
		id: {
			type: Number,
			required: true,
		},
	});

	const { loadShowEpisodes } = useFetchShows();
	const {
		data: episodes,
		isFetching,
		error: ErrorText,
	} = loadShowEpisodes(props.id);
</script>
<template>
	<v-list :height="300">
		<v-list-item
			v-if="episodes"
			v-for="(episode, i) in episodes"
			:key="i"
			@click="() => openExternalURL(episode.url)"
			:prepend-avatar="imgURL(episode.image?.medium)"
			class="pa-5"
			elevation="1"
		>
			<template v-if="episode.name" v-slot:title>
				{{ episode.name }}
			</template>
			<template v-if="episode.number" v-slot:subtitle>
				Episode {{ episode.number }}
			</template>
			<template v-slot:append>
				<v-icon>mdi-play</v-icon>
			</template>
		</v-list-item>
		<Loader v-else-if="isFetching" />
		<div class="d-flex justify-center align-center" style="height: 100%" v-else>
			<Error :text="ErrorText" />
		</div>
	</v-list>
</template>
