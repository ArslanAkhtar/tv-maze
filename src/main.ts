import './assets/main.css'

import { createApp } from 'vue'
import App from './App.vue'
import router from './router'
import 'vuetify/styles'
import { createVuetify } from 'vuetify'
import * as components from 'vuetify/components'
import * as directives from 'vuetify/directives'
import { fa } from 'vuetify/iconsets/fa'
import { mdi, aliases } from 'vuetify/iconsets/mdi'

import '@mdi/font/css/materialdesignicons.css'

const app = createApp(App)

app.use(router)

const vuetify = createVuetify({
  components,
  directives,
  theme: {
    defaultTheme: 'dark'
  },
  icons: {
    defaultSet: 'mdi',
    aliases,
    sets: {
      mdi,
      fa
    }
  }
})

app.use(vuetify)

app.mount('#app')
