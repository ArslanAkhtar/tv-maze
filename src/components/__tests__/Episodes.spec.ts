import { describe, it, expect, vi } from "vitest";
import useFetchShows from "@/composables/FetchData";
import Episodes from "../Episodes.vue";
import { mount } from "@vue/test-utils";
import { vuetify } from "@/main";
import expectedAPIResponse from "./mockData/mockEpisode.json";
import type { ExtVueWrapper } from "@/helpers/types";

describe("Episodes Component", () => {
	const id = 1;
	const wrapper = mount(Episodes, {
		props: { id: id },
		global: {
			components: {
				Episodes,
			},
		},
		plugins: [vuetify],
	});
	it("matches data from api based on id prop", async () => {
		const { loadShowEpisodes } = useFetchShows();
		const { data } = await loadShowEpisodes(wrapper.props().id);
		expect(data.value[0]).toMatchObject(expectedAPIResponse);
	});

	test("renders correctly", () => {
		expect(wrapper.element).toMatchSnapshot();
	});

	it("opens URL in a new tab", async () => {
		const url = "https://example.com";
		vi.spyOn(window, "open");
		const res = wrapper.vm as unknown as ExtVueWrapper;
		await res.openURL(url);
		expect(window.open).toHaveBeenCalledWith(url, "_blank");
	});

	it("returns the input URL if provided", () => {
		const inputURL = "https://example.com/image.jpg";

		const res = wrapper.vm as unknown as ExtVueWrapper;

		const result = res.imgURL(inputURL);

		expect(result).toBe(inputURL);
	});
});
