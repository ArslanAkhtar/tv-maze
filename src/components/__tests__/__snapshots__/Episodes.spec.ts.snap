// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Episodes Component > renders correctly 1`] = `
<v-list
  height="300"
>
  
  <v-list-item
    class="pa-5"
    elevation="1"
    prepend-avatar="https://static.tvmaze.com/uploads/images/medium_landscape/1/4388.jpg"
  />
  <v-list-item
    class="pa-5"
    elevation="1"
    prepend-avatar="https://static.tvmaze.com/uploads/images/medium_landscape/1/4389.jpg"
  />
  <v-list-item
    class="pa-5"
    elevation="1"
    prepend-avatar="https://static.tvmaze.com/uploads/images/medium_landscape/1/4390.jpg"
  />
  <v-list-item
    class="pa-5"
    elevation="1"
    prepend-avatar="https://static.tvmaze.com/uploads/images/medium_landscape/1/4391.jpg"
  />
  <v-list-item
    class="pa-5"
    elevation="1"
    prepend-avatar="https://static.tvmaze.com/uploads/images/medium_landscape/1/4392.jpg"
  />
  <v-list-item
    class="pa-5"
    elevation="1"
    prepend-avatar="https://static.tvmaze.com/uploads/images/medium_landscape/1/4393.jpg"
  />
  <v-list-item
    class="pa-5"
    elevation="1"
    prepend-avatar="https://static.tvmaze.com/uploads/images/medium_landscape/1/4394.jpg"
  />
  <v-list-item
    class="pa-5"
    elevation="1"
    prepend-avatar="https://static.tvmaze.com/uploads/images/medium_landscape/1/4395.jpg"
  />
  <v-list-item
    class="pa-5"
    elevation="1"
    prepend-avatar="https://static.tvmaze.com/uploads/images/medium_landscape/1/4396.jpg"
  />
  <v-list-item
    class="pa-5"
    elevation="1"
    prepend-avatar="https://static.tvmaze.com/uploads/images/medium_landscape/1/4397.jpg"
  />
  <v-list-item
    class="pa-5"
    elevation="1"
    prepend-avatar="https://static.tvmaze.com/uploads/images/medium_landscape/1/4398.jpg"
  />
  <v-list-item
    class="pa-5"
    elevation="1"
    prepend-avatar="https://static.tvmaze.com/uploads/images/medium_landscape/1/4399.jpg"
  />
  <v-list-item
    class="pa-5"
    elevation="1"
    prepend-avatar="https://static.tvmaze.com/uploads/images/medium_landscape/1/4400.jpg"
  />
  
</v-list>
`;
